### ISSUE TITLE
Update S1481 (`sonar-no-unused-vars`): cover cases from ESLint's implementation

### ISSUE THREAD

**Issue by ilia-kebets-sonarsource** [SonarSource/SonarJS#3746]
- `try{}catch(err){}`
- `const data = { type: 'yolo', x: 2, y: 2 };\nconst { type, ...unused } = data;\n console.log(type)` (var is initialized, decomposed somehow and its decomposition is not (fully) used)

**Comment by ilia-kebets-sonarsource** [comment]
actually, we don't flag unused function params or error params from catch clauses: https://github.com/SonarSource/SonarJS/blob/4bae3f88d1af35f7606571d18c099d0741155fac/tests/linting/eslint/rules/sonar-no-unused-vars.test.ts#L41

**Comment by ilia-kebets-sonarsource** [comment]
The other use case is covered if it's local, aka inside a function:

```javascript
      function foo () {
        const data = {
          type: 'yolo',
          x: 2,
          y: 2
        };
        const { type, ...unused } = data;
        console.log(type);
      }
```

raises an issue on (8,26)
